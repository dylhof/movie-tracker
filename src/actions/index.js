export const storeMovies = (movies) => ({
  type: 'STORE_MOVIES',
  movies
})

export const setCurrentUser = (name, id) => ({
  type: 'SET_USER',
  name,
  id
})

export const logoutCurrentUser = () => ({
  type: 'LOGOUT_USER'
})

export const addFavorite = (id) => ({
  type: 'ADD_FAVORITE',
  id
})

export const addAllUserFavorites = (favorites) => ({
  type: 'ADD_ALL_USER_FAVORITES',
  favorites
})

export const deleteFavorite = (id) => ({
  type: 'DELETE_FAVORITE',
  id
})

export const clearFavorites = () => ({
  type: 'CLEAR_FAVORITES',
})

export const setLoading = (bool) => ({
  type: 'SET_LOADING',
  bool
})