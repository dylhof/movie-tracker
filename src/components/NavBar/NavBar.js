import React from 'react';
import { NavLink } from 'react-router-dom';

export const NavBar = () => {
  const member = false;
  return (
    <div className="NavBar">
      <NavLink exact to="/">Home</NavLink>
      <NavLink exact to={(member) ? '/Favorites' : '/Login'}>
        Favorites
      </NavLink>
      <NavLink exact to='/Login'>Login</NavLink>
      <NavLink exact to='/SignUp'>Sign up</NavLink>
      {/* this navlink will take you to a new login page*/}
    </div>
  );
}