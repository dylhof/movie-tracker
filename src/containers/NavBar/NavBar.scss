.NavBar {
  display: flex;
  flex-direction: column;
  align-items: center;

}
.home-btn {
  width: 395px;
  height: 87px;
  font-size: 53px;
  margin: 15px;
  @include navbutton();
  
}

.small-btns {
  display: flex;
  justify-content: space-between;
  width: 405px;
  
  .signup-login-btns {
    display: flex;
    justify-content: space-between;
    width: 66%;
  }
}

.favorites-btn,
.login-btn,
.signup-btn,
.logout-btn {
  height: 43px;
  @include navbutton();
}

.favorites-btn,
.logout-btn {
  width: 114px;
}

.login-btn,
.signup-btn {
 width: 114px;
}

.logout-btn {
  &:hover {
    cursor: pointer;
  }
}

.nav-link {
  text-decoration: none;
}