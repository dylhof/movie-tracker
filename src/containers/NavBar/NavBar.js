import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux'
import { logoutCurrentUser, clearFavorites } from '../../actions';

class NavBar extends Component {
  // constructor() {
  //   super()
  // }

  handleLogoutClick = () =>{
    this.props.dispatchLogoutUser()
    this.props.dispatchClearFavorites()
  }
  render() {
    return (
      <div className="NavBar">
        <NavLink exact to="/">Home</NavLink>
        <NavLink exact to={(this.props.currentUser) ? '/favorites' : '/login'}>
          Favorites
      </NavLink>
        {
          this.props.currentUser ?
            <button onClick={this.handleLogoutClick}>Logout</button>
            :
            <div>
              <NavLink exact to='/login'>Login</NavLink>
              <NavLink exact to='/signUp'>Sign up</NavLink>
            </div>
        }
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  dispatchLogoutUser: () => dispatch(logoutCurrentUser()),
  dispatchClearFavorites: () => dispatch(clearFavorites())
})

export const mapStateToProps = (state) => ({
  currentUser: state.currentUser
})

export default connect(mapStateToProps, mapDispatchToProps)(NavBar)