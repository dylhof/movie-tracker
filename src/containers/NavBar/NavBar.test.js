import React from 'react'
import { shallow } from 'enzyme'
import { NavBar, mapDispatchToProps, mapStateToProps } from './NavBar';
import * as actions from '../../actions/index'

describe('NavBar', () => {
  let wrapper
  beforeEach(() => {
    const mockDispatchLogoutUser = jest.fn()
    const mockDispatchClearFavorites = jest.fn()
    const mockCurrentUser = {name: 'Dylan', id: 1}
    wrapper = shallow(
      <NavBar 
        dispatchLogoutUser={mockDispatchLogoutUser}
        dispatchClearFavorites={mockDispatchClearFavorites}
        currentUser={mockCurrentUser}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot with no current user', () => {
    wrapper = shallow(
      <NavBar currentUser={null} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  // it('should call handleLogoutClick', () => {
  //   wrapper.instance().handleLogoutClick = jest.fn()
  //   wrapper.find('button').simulate('click')
  //   expect(wrapper.instance().handleLogoutClick).toHaveBeenCalled()
  // })

  describe('handleLogoutClick', () => {
    it('should call dispatchLogoutUser and dispatchClearFavorites', () => {
      //execution
      wrapper.instance().handleLogoutClick()
      //expectation
      expect(wrapper.instance().props.dispatchLogoutUser).toHaveBeenCalled()
      expect(wrapper.instance().props.dispatchClearFavorites).toHaveBeenCalled()
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using dispatchLogoutUser from MDTP', () => {
      //setup
      const mockDispatch = jest.fn()
      const actionToDispatch = actions.logoutCurrentUser()
      //execution
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.dispatchLogoutUser()
      //expectation
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should call dispatch when using dispatchClearFavorites from MDTP', () => {
      //setup
      const mockDispatch = jest.fn()
      const actionToDispatch = actions.clearFavorites()
      //execution
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.dispatchClearFavorites()
      //expectation
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })

  describe('mapStateToProps', () => {
    it('should return an object with a current user', () => {
      //setup
      const mockState = {currentUser: {name: 'Matt'}, favorites: [1, 4]}
      const expected = {currentUser: {name: 'Matt'}}
      //execution
      const mappedProps = mapStateToProps(mockState)
      //expectation
      expect(mappedProps).toEqual(expected)
    })
  })

})