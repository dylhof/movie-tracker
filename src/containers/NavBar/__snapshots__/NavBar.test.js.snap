// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match the snapshot 1`] = `
<div
  className="NavBar"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav-link"
    exact={true}
    to="/"
  >
    <div
      className="home-btn"
    >
      Movie Tracker
    </div>
  </NavLink>
  <div
    className="small-btns"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="nav-link"
      exact={true}
      to="/favorites"
    >
      <div
        className="favorites-btn"
      >
        Favorites
      </div>
    </NavLink>
    <div
      className="logout-btn"
      onClick={[Function]}
    >
      Logout
    </div>
  </div>
</div>
`;

exports[`NavBar should match the snapshot with no current user 1`] = `
<div
  className="NavBar"
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav-link"
    exact={true}
    to="/"
  >
    <div
      className="home-btn"
    >
      Movie Tracker
    </div>
  </NavLink>
  <div
    className="small-btns"
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="nav-link"
      exact={true}
      to="/login"
    >
      <div
        className="favorites-btn"
      >
        Favorites
      </div>
    </NavLink>
    <div
      className="signup-login-btns"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="nav-link"
        exact={true}
        to="/login"
      >
        <div
          className="login-btn"
        >
          Login
        </div>
      </NavLink>
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="nav-link"
        exact={true}
        to="/signUp"
      >
        <div
          className="signup-btn"
        >
          Sign up
        </div>
      </NavLink>
    </div>
  </div>
</div>
`;
